Normalize domains to lowercase when comparing URLs

bamarni opened the pull request:
Hostnames are case-insensitive and this caused issues with the `dcos
package install` command and CLI resources, as the auth token wasn't
being sent when the configured cluster URL and CLI resource URL had a
case mismatch on the domain.

https://jira.mesosphere.com/browse/DCOS_OSS-4501
https://jira.mesosphere.com/browse/DCOS-45391

armandgrillet commented:
The AppVeyor issue, `The build phase is set to "MSBuild" mode (default), but no Visual Studio project or solution files were found in the root directory. If you are not building Visual Studio project switch build mode to "Script" and provide your custom build command.` has already been seen before.

bamarni committed:
Normalize domains to lowercase when comparing URLs

Hostnames are case-insensitive and this caused issues with the `dcos
package install` command and CLI resources, as the auth token wasn't
being sent when the configured cluster URL and CLI resource URL had a
case mismatch on the domain.

https://jira.mesosphere.com/browse/DCOS_OSS-4501
https://jira.mesosphere.com/browse/DCOS-45391
